base/common/wide_integer_impl.h

```diff
@@ -239,6 +239,13 @@ struct integer<Bits, Signed>::_impl
     template <class T>
     constexpr static void set_multiplier(integer<Bits, Signed> & self, T t) noexcept {
         constexpr uint64_t max_int = std::numeric_limits<uint64_t>::max();
+
+        /// Replacing nan to zero, and inf to max(T) to avoid stack overflow in bigint_cast
+        if (isnan(t))
+            t = 0;
+        if (isinf(t))
```

Comment:
There is also -inf that probably better to be replaced to the min value.
But as it's implementation specific behaviour, let's replace all to zero.